CPS-XXXX/README.md

```diff
@@ -0,0 +1,91 @@
+---
+CPS: ?
+Title: Full determinism of transactions is unnecessarily restrictive for DeFi
+Status: Open
+Category: Ledger
+Authors:
+    - Vladimir Kalnitsky <klntsky@gmail.com>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/?
+Created: YYYY-MM-DD
+License: CC-BY-4.0
+---
+
+<!-- Existing categories:
+
+- Meta     | For meta-CIPs which typically serves another category or group of categories.
+- Wallets  | For standardisation across wallets (hardware, full-node or light).
+- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+- Metadata | For proposals around metadata (on-chain or off-chain).
+- Tools    | A broad category for ecosystem tools not falling into any other category.
+- Plutus   | Changes or additions to Plutus
+- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+-->
+
+## Abstract
+
+Determinism of transactions gives rise to valuable ledger properties, such as:
+
+- atomicity (a transaction is either fully accepted or fully rejected without a fee)
+- predictability of monetary changes and fees
+
+However, it comes at a cost:
+
+- Development becomes more complicated because of the need to "re-invent" non-determinism where it is required for the dApp logic
+- UTxO contention limits the number of concurrent users a dApp can have
```

Comment:
> Each seller can have their own UTxO which makes concurrency as high as the underlying market

The problem is that each buyer will try to use the order with the best price available. No matter how many UTxOs you create in an order book, it will be a probabilistic game without batchers, but it doesn't have to be in principle. By parallelizing you can only sidestep the problem, not solve it, which is not really viable, as EVM-based AMMs can just provide better guarantees: they can offer immediate execution no matter the number of users. Aiming at something less than that means losing, because the quality standards are not being set by you